validation/forms.py

```diff
@@ -0,0 +1,62 @@
+from django import forms
+import re
+from django.core.exceptions import ObjectDoesNotExist
+from django.contrib.auth.models import User
+
+
+class Login(forms.Form):
+    username = forms.CharField(
+        required=True,
+        widget=forms.TextInput(
+            attrs={"class": "form-control", "style": "max-width: 15rem;"}
+        ),
+    )
+    password = forms.CharField(
+        required=True,
+        widget=forms.PasswordInput(
+            attrs={"class": "form-control", "style": "max-width: 15rem;"}
+        ),
+    )
+
+
+class Register(forms.Form):
+    first_name = forms.CharField(
+        required=True,
+        widget=forms.TextInput(
+            attrs={"class": "form-control", "style": "max-width: 15rem;"}
+        ),
+    )
+    last_name = forms.CharField(
+        required=True,
+        widget=forms.TextInput(
+            attrs={"class": "form-control", "style": "max-width: 15rem;"}
+        ),
+    )
+    username = forms.CharField(
+        required=True,
+        widget=forms.TextInput(
+            attrs={"class": "form-control", "style": "max-width: 15rem;"}
+        ),
+    )
+    password = forms.CharField(
+        required=True,
+        widget=forms.PasswordInput(
+            attrs={"class": "form-control", "style": "max-width: 15rem;"}
+        ),
+    )
+
+    def clean_username(self):
+        username = self.cleaned_data["username"]
+        if not re.search(
+            r"^\w+$", username
```

Comment:
I had to look this up, but `\w` matches alphanumerics in any Unicode script, e.g., `î४`. That’s probably fine? I just found it interesting.

Reply:
I'm not sure that we have any requirements for username and I highly doubt our users will be using special characters in their usernames. I do wonder if this would cause issues in how the usernames are stored.